lib/chef/application.rb

```diff
@@ -151,6 +152,20 @@ def configure_logging
       Chef::Application.fatal!("Aborting due to invalid 'log_location' configuration", 2)
     end
 
+    # Turn `log_location :syslog` and `log_location :win_evt` into the
+    # appropriate loggers.
+    def configure_log_location
+      log_location = Chef::Config[:log_location]
+      return unless log_location.respond_to?(:to_sym)
```

Comment:
No, this just returns early when `Chef::Config[:log_location]` isn't a string or symbol, so if it's e.g. STDIN or a Logger object it won't be changed.